spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -231,6 +231,17 @@ class OptimizeExecutor(
     optimizeContext: DeltaOptimizeContext)
   extends DeltaCommand with SQLMetricsReporting with Serializable {
 
+  /**
+   * In which mode the Optimize command is running. There are three valid modes:
+   * 1. Compaction
+   * 2. ZOrder
+   * 3. Clustering
+   */
+  private val optimizeTableMode = OptimizeTableStrategy.getMode(txn.snapshot, zOrderByColumns)
+
+  private val optimizeStrategy = OptimizeTableStrategy(
+    optimizeTableMode, sparkSession, txn.snapshot, optimizeContext, zOrderByColumns)
+
   /** Timestamp to use in [[FileAction]] */
   private val operationTimestamp = new SystemClock().getTimeMillis()
 
```

Comment:
I think we can remove `isClusteredTable` and `isMultiDimClustering`?

Reply:
`isMultiDimClustering ` is still needed for places such as creating `repartitionDF` in `runOptimizeBinJob`. I started hiding them as well in the strategy class but that made the strategy class much more complex since it has more denepency from the Optimize table command's context.

The same goes to `isClusteredTable `.